### PR TITLE
Fix Need refresh the complete page to use workflows in content types …

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
…portlet

### CODE REVIEW

File: src/app/api/services/dot-license/dot-license.service.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
+/**
+ * Handle license information of current logged in user
+ * @export
+ * @class DotLicenseService
+ */
 @Injectable()
 export class DotLicenseService {
-    constructor(private dotcmsConfig: DotcmsConfig) {}
+    private configUrl: string;
```
Comment (freddyucv) [inline src/app/api/services/dot-license/dot-license.service.ts]: I think here is better a name like: licenseURL, after all we should change this url by the license end point url when we'll have it

File: src/app/api/services/dot-license/dot-license.service.ts [hunk 2]
```diff
@@ -13,10 +23,12 @@ export class DotLicenseService {
...
     isEnterpriseLicense(): Observable<boolean> {
-        return this.dotcmsConfig
-            .getConfig()
-            .take(1)
-            .pluck('license')
+        return this.coreWebService
+            .requestView({
+                method: RequestMethod.Get,
+                url: this.configUrl
+            })
```
Comment (freddyucv) [inline src/app/api/services/dot-license/dot-license.service.ts]: Maybe we should have a 'getLicense' method here and do into it the request (maybe it should be private at least for now), later we could change the 'getLicense' method for not do a request every time when we'll have a web socket event

File: src/app/api/services/dot-license/dot-license.service.spec.ts [hunk 3]
```diff
@@ -1,80 +1,85 @@
...
+                                    level: 400
+                                }
+                            }
+                        }
+                    }
+                })
+            )
         );
 
-        dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
```
Comment (freddyucv) [inline src/app/api/services/dot-license/dot-license.service.spec.ts]: maybe for we should have a test to check that a request is sent to the BE each time

File: src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts [hunk 4]
```diff
@@ -44,6 +45,7 @@ describe('DotEditPageMainComponent', () => {
                 providers: [
                     { provide: DotMessageService, useValue: messageServiceMock },
                     { provide: PageViewService, useClass: PageViewServiceMock },
+                    DotLicenseService
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts]: why?

Comment (hmoreras) [reply src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts]: mock it ? or add DotLicenseService to DotTestBed ?

File: src/app/api/services/dot-license/dot-license.service.spec.ts (hunk 3 above)
Comment (fmontes) [reply src/app/api/services/dot-license/dot-license.service.spec.ts]: yup

File: src/app/api/services/dot-license/dot-license.service.spec.ts [hunk 5]
```diff
@@ -1,80 +1,85 @@
...
 
     it('should return a false response because license is 100 = Community', () => {
-        let result: boolean;
-        dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        let result;
+        this.dotLicenseService.isEnterpriseLicense().subscribe((res) => (result = res));
+        this.lastConnection.mockRespond(
+            new Response(
+                new ResponseOptions({
+                    body: {
```
Comment (fmontes) [inline src/app/api/services/dot-license/dot-license.service.spec.ts]: You might want to create a util that you send the level and return you the object, to many duplicated code to mock this.

File: src/app/api/services/dot-license/dot-license.service.ts [hunk 6]
```diff
@@ -1,10 +1,20 @@
...
+ * @export
+ * @class DotLicenseService
+ */
 @Injectable()
 export class DotLicenseService {
-    constructor(private dotcmsConfig: DotcmsConfig) {}
+    private licenseURL: string;
+
+    constructor(private coreWebService: CoreWebService) {
+        this.licenseURL = 'v1/appconfiguration';
```
Comment (fmontes) [inline src/app/api/services/dot-license/dot-license.service.ts]: Why don't initialize this in the declaration?

File: src/app/portlets/content-types/form/content-types-form.component.ts [hunk 7]
```diff
@@ -269,15 +268,13 @@ export class ContentTypesFormComponent implements OnInit {
...
-            .filter((workflow: DotWorkflow) => workflow.system)
-            .map((workflow: DotWorkflow) => workflow.id);
+        const defaultValue = workflows.filter((workflow: DotWorkflow) => workflow.system).map((workflow: DotWorkflow) => workflow.id);
 
         this.form.get('workflow').setValue(defaultValue);
     }
 
-    private updateWorkflowFormControl(license): void {
-        if (!license.isCommunity) {
+    private updateWorkflowFormControl(isEnterpriseLicense): void {
```
Comment (fmontes) [inline src/app/portlets/content-types/form/content-types-form.component.ts]: type

File: src/app/portlets/content-types/main/content-types.component.spec.ts [hunk 8]
```diff
@@ -69,7 +69,7 @@ describe('ContentTypesPortletComponent', () => {
             'contenttypes.fieldname.entries': 'Entries',
             'contenttypes.fieldname.structure.name': 'Content Type Name',
             'contenttypes.content.variable': 'Variable Name',
-            'mod_date': 'Last Edit Date',
+            mod_date: 'Last Edit Date',
```
Comment (fmontes) [inline src/app/portlets/content-types/main/content-types.component.spec.ts]: ?

File: src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts (hunk 4 above)
Comment (fmontes) [reply src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts]: DotTestBed

File: src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts [hunk 9]
```diff
@@ -15,7 +15,7 @@ import { DOTTestBed } from '../../../../test/dot-test-bed';
 import { DotEditPageNavComponent } from '../dot-edit-page-nav/dot-edit-page-nav.component';
 import { PageViewServiceMock } from '../../../../test/page-view.mock';
 import { DotRenderedPageState } from '../../shared/models/dot-rendered-page-state.model';
-import { DotLicenseService } from '../../../../api/services/dot-license/dot-license.service';
+import { debug } from 'util';
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/main/dot-edit-page-main/dot-edit-page-main.component.spec.ts]: ?

**Pull request merged**